frontend/packages/schema-editor/src/components/SchemaInspector/ItemRestrictions/ItemRestrictions.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { isField, isReference, pointerIsDefinition, UiSchemaNode } from '@altinn/schema-model';
+import { FieldType } from '@altinn/schema-model';
+import { setRequired, setRestriction, setRestrictions } from '@altinn/schema-model';
+import { ArrayRestrictions } from './ArrayRestrictions';
+import { NumberRestrictions } from './NumberRestrictions';
+import { ObjectRestrictions } from './ObjectRestrictions';
+import { StringRestrictions } from './StringRestrictions';
+import classes from './ItemRestrictions.module.css';
+import { Switch } from '@digdir/design-system-react';
+import { Divider } from 'app-shared/primitives';
+import { useTranslation } from 'react-i18next';
+import { KeyValuePairs } from 'app-shared/types/KeyValuePairs';
+import { useSchemaEditorAppContext } from '@altinn/schema-editor/hooks/useSchemaEditorAppContext';
+import { EnumList } from './EnumList';
+
+export interface RestrictionItemProps {
+  restrictions: any;
+  readonly: boolean;
+  path: string;
+  onChangeRestrictionValue: (id: string, key: string, value?: string | boolean) => void;
+  onChangeRestrictions: (id: string, restrictions: KeyValuePairs) => void;
+}
+
+export type ItemRestrictionsProps = {
+  schemaNode: UiSchemaNode;
+};
+
+export const ItemRestrictions = ({ schemaNode }: ItemRestrictionsProps) => {
+  const { pointer, isRequired, isArray, restrictions } = schemaNode;
+  const { schemaModel, save } = useSchemaEditorAppContext();
+
+  const handleRequiredChanged = (e: any) => {
```

Comment:
Should we type the checkbox event to `ChangeEvent` instead of any? 😊 
```suggestion
  const handleRequiredChanged = (event: ChangeEvent<HTMLCheckboxElement>) => {
```

Reply:
I believe it should be ```HTMLInputElement``` as it seems like ```HTMLCheckboxElement``` does not exist?